source/adios2/ADIOSTypes.h

```diff
@@ -137,6 +114,35 @@ struct TypeInfo<T, typename std::enable_if<std::is_same<
     using ValueType = typename T::value_type;
 };
 
+const size_t UnknownDim = 0;
+const size_t JoinedDim = SIZE_MAX - 1;
```

Comment:
the problem with the std is that it doesn't provide compile-time limits. Just like vector.max_size(), numeric_limits only provides runtime solutions. So far,  (size_t )-1 does the job at compile time and it's portable.